testing/src/main/java/io/grpc/internal/testing/StatsTestUtils.java

```diff
@@ -93,39 +107,29 @@ public long getMetricAsLongOrFail(MeasurementDescriptor metricName) {
   }
 
   /**
-   * This tag will be propagated by {@link FakeStatsContextFactory} on the wire.
+   * This tag will be propagated by {@link FakeTagger} on the wire.
    */
   public static final TagKey EXTRA_TAG = TagKey.create("/rpc/test/extratag");
 
   private static final String EXTRA_TAG_HEADER_VALUE_PREFIX = "extratag:";
   private static final String NO_EXTRA_TAG_HEADER_VALUE_PREFIX = "noextratag";
```

Comment:
I wonder if this and its related logic can be removed.
It was needed because the old StatsContext didn't provide a way to tell if it's empty, thus the metadata was always sent. Now that you have implemented the optimization to skip sending metdata if TagContext is empty, we should no longer need the metadata form that represents "no tag".

Reply:
I'm not sure I understand.  I didn't try to change any of the logic for serializing and deserializing tags.

Reply:
I meant if you remove this field and the if-branches that references this field, tests should not fail. Because of the following `if` you added to `CensusStatsModule`, those branches will never be executed now.
```java
        if (!module.tagger.empty().equals(parentCtx)) {
          headers.put(module.statsHeader, parentCtx);
        }
```

Reply:
You are right ... the optimization was added before.

To keep it simple, `FakeTagContextBinarySerializer` is supposed to only handle `EXTRA_TAG`, and tests are supposed to only add `EXTRA_TAG`. Now `fromByteArray()` will fail if the metadata contains anything other than `EXTRA_TAG`. `toByteArray()` will fail too, by throwing NullPointerException -- it's probably better to make it throw `UnsupportedOperationException` if `extraTagValue==null`.

Reply:
I made toByteArray throw an UnsupportedOperationException.